packages/lesswrong/lib/vulcan-lib/components.tsx

```diff
@@ -46,67 +46,36 @@ const coreComponents = [
 type C<T=any> = React.ComponentType<T>
 type HoC<O,T> = (component: C<O>) => C<T>
 
-/**
- * Register a Vulcan component with a name, a raw component than can be extended
- * and one or more optional higher order components.
- *
- * @param {String} name The name of the component to register.
- * @param {React Component} rawComponent Interchangeable/extendable component.
- * @param {...Function} hocs The HOCs to compose with the raw component.
- *
- * Note: when a component is registered without higher order component, `hocs` will be
- * an empty array, and it's ok!
- * See https://github.com/reactjs/redux/blob/master/src/compose.js#L13-L15
- *
- * @returns Structure of a component in the list:
- *
- * ComponentsTable.Foo = {
- *    name: 'Foo',
- *    hocs: [fn1, fn2],
- *    rawComponent: React.Component,
- *    call: () => compose(...hocs)(rawComponent),
- * }
- *
- */
-/*export function registerComponent<PropType>(name, rawComponent, ...hocs): React.ComponentType<Omit<PropType,"classes">> {
-  var styles = null;
-  
-  // support single-argument syntax
-  if (typeof arguments[0] === 'object') {
-    // note: cannot use `const` because name, components, hocs are already defined
-    // as arguments so destructuring cannot work
-    // eslint-disable-next-line no-redeclare
-    var { name, component, hocs = [], styles } = arguments[0];
-    rawComponent = component;
-  } else if (typeof arguments[2] === 'object' && (arguments[2].hocs || arguments[2].styles)) {
-    styles = arguments[2].styles;
-    hocs = arguments[2].hocs || [];
-  }
-  
-  if (styles) {
-    hocs.push(withStyles(styles, {name: name}));
-  }
-  
-  rawComponent.displayName = name;
-  
-  if (name in ComponentsTable && ComponentsTable[name].rawComponent !== rawComponent) {
-    throw new Error(`Two components with the same name: ${name}`);
+const addClassnames = (componentName: string) => {
+  const classesProxy = new Proxy({}, {
+    get: function(obj: any, prop: any) {
```

Comment:
Shouldn't `prop` be a string here?

Reply:
In principle it could be something else, if someone were to were `classes[not_a_string]`.